snow/engine/snowman/engine.go

```diff
@@ -190,12 +193,19 @@ func (e *Engine) Gossip(ctx context.Context) error {
 		return nil
 	}
 
+	blkID := e.Consensus.Preference()
+
+	if e.abortDueToInsufficientConnectedStake(blkID) {
+		return nil
+	}
```

Comment:
I'm not sure that this change is really needed / relevant. The `PullQuery` sent here is used just to learn about potential new blocks (it can't impact the snowball preference strength).

Was it intended to disable fetching new blocks when not connected to sufficient stake as well?

Reply:
But doesn't handling of the PullQuery cause chits to be [sent](https://github.com/ava-labs/avalanchego/blob/a922d0f5f03821dc5c032dbd66006a7f46259c8c/snow/engine/snowman/engine.go#L312), which in turn causes snowman [voting](https://github.com/ava-labs/avalanchego/blob/a922d0f5f03821dc5c032dbd66006a7f46259c8c/snow/engine/snowman/engine.go#L400)?

Reply:
The chit gets dropped here: https://github.com/ava-labs/avalanchego/blob/a922d0f5f03821dc5c032dbd66006a7f46259c8c/snow/consensus/snowman/poll/set.go#L121 because we never register a poll for this requestID here

Reply:
oh, I see.

Even though a node that has sent a pull query is responded to with chits, it is only if the pull query originated from [sendQuery](https://github.com/ava-labs/avalanchego/blob/a922d0f5f03821dc5c032dbd66006a7f46259c8c/snow/engine/snowman/engine.go#L916) that it can actually get into snowman.